libs/api/content-search/src/services/elastic.service.ts

```diff
@@ -77,15 +78,26 @@ export class ElasticService {
     // if we have any requests execute them
     if (requests.length) {
       try {
+        const chunkSize = 200 // this has to be an even number
         const client = await this.getClient()
-        const response = await client.bulk({
-          index: index,
-          body: requests,
-        })
-        // TODO: ES errors while indexing might not throw, but appear in response log those errors here
+        let requestChunk = requests.splice(-chunkSize, chunkSize)
+        do {
+          // wait for request b4 continuing
+          const response = await client.bulk({
+            index: index,
+            body: requestChunk,
+          })
+          if (response.body.errors) {
+            logger.error('Failed to import some documents in bulk import', {
+              response,
+            })
+          }
+          requestChunk = requests.splice(-chunkSize, chunkSize)
+        } while (requestChunk.length) // while we have requests to process
```

Comment:
this is a hard to read code
"do while", instead of regular while; which would make the "if (requests.length)" redundant and minimize nested code (i.e. the indentation)
or building a list of chunks and doing a forEach on that, which I think would be more readable.